src/averageAges.js

```diff
@@ -36,8 +43,20 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = null) {
+  let women = people.filter(person => person.sex === 'f');
+
+  if (withChildren !== null) {
+    women = people
```

Comment:
You should filter the 'women' array instead of the 'people' array when checking for women with children.